src/features/quest/ui/styles.ts

```diff
@@ -0,0 +1,221 @@
+import styled, { css } from 'styled-components';
+import { getImageUrl } from '@utils/getImageUrl';
+import { media } from '@/style/media';
+
+interface QuestUrlProps {
+  $isLearn?: boolean;
+  $isQuest?: boolean;
+}
+
+export const QuestContainer = styled.div<QuestUrlProps>`
+  ${({ $isLearn }) =>
+    $isLearn &&
+    css`
+      width: 270px;
+      margin-right: 28px;
+
+      ${media.mobile} {
+        display: none;
+      }
+    `}
+
+  ${({ $isQuest = true }) =>
+    $isQuest &&
+    css`
+      width: 683px;
+      margin-top: 40px;
+    `}
+`;
```

Comment:
```suggestion
const questContainerLearn = css`
    width: 270px;
    margin-right: 28px;
    ${media.mobile} {
      display: none;
    }
`

const qeustContainerQuest = css`
     width: 683px;
     margin-top: 40px;
`

export const QuestContainer = styled.div<QuestUrlProps>`
    ${({ $isLearn }) => $isLearn && questContainerLearn;
    ${({ $isQuest }) => $isQuest && questContaienrQuest;
`
```
뭔가 css와 조건부 로직을 분리해 봤는데, 생각보다 별로인 것 같네요.
처음에는 `css`가 분리가 되니깐 더 보기 편하지 않을까 싶었는데. 좀 애매한 것 같습니다.

Reply:
### 기존 코드
```tsx
export const ProgressBarWrapper = styled.div<QuestUrlProps>`
  display: flex;
  justify-content: center;
  align-items: flex-end;

  ${({ $isLearn }) =>
    $isLearn &&
    css`
      gap: 3px;
    `}
  ${({ $isQuest = true }) =>
    $isQuest &&
    css`
      gap: 15px;
    `}
`;
```
### 수정 코드
```tsx
const progressBarWrapperLearn = css`
  gap: 3px;
`;

const progressBarWrapperQuest = css`
  gap: 15px;
`;

export const ProgressBarWrapper = styled.div<QuestUrlProps>`
  display: flex;
  justify-content: center;
  align-items: flex-end;

  ${({ $isLearn }) => $isLearn && progressBarWrapperLearn}
  ${({ $isQuest = true }) => $isQuest && progressBarWrapperQuest}
`;
```
리뷰 감사합니다! 말씀하신 것처럼 위와 같이 변경해봤는데 가독성도 좋고 생각해보니 변수 부분만 따로 테스트 해볼 수 있어서 좋을 것 같네요!

Reply:
하나 궁금한 게 있습니다. 혹시 변수 부분을 위로 다 빼고 스타일드 컴포넌트를 그 밑에 다 빼는 게 좋을까요?? 아니면 (동일한)변수와 스타일드 컴포넌트를 하나로 묶어서 하는 게 좋을까요? 좀 고민됩니다... 변수에다가 주석을 따로 달아야 할지도 고민되고요..
```tsx
// 스타일 변수
const questContainerLearn = css`
  width: 270px;
  margin-right: 28px;

  ${media.mobile} {
    display: none;
  }
`;

const questContainerQuest = css`
  width: 683px;
  margin-top: 40px;
`;

const questSectionLearn = css`
  width: 270px;
  height: 210px;
  margin: 89px 0 0 0;
  position: fixed;
  background-image: url(${getImageUrl('종이.svg')});
`;

const questContentWrapperQuest = css`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: flex-start;
`;

// 스타일드 컴포넌트
export const QuestContainer = styled.div<QuestUrlProps>`
  ${({ $isLearn }) => $isLearn && questContainerLearn}
  ${({ $isQuest = true }) => $isQuest && questContainerQuest}
`;

export const QuestSection = styled.section<Omit<QuestUrlProps, '$isQuest'>>`
  ${({ $isLearn }) => $isLearn && questSectionLearn}
`;

export const QuestContentWrapper = styled.div<Omit<QuestUrlProps, '$isLearn'>>`
  ${({ $isQuest }) => $isQuest && questContentWrapperQuest}
`;
```

Reply:
@bluetree7878 변수명을 명확하게 한다면, 차라리 분리하는게 어떨까요?🧐
물론 제 생각이긴 합니다. 근데, 실제로 사용하는 부분은 `Styled-components`를 사용하기 때문에 같이 두는 것 또한 좋은 선택지 일 수 있을 것 같네요🥲 어렵습니다...

이 부분은 팀원과 상의하여 진행하는게 좋을 듯 싶습니다.😅

Reply:
@zzzRYT 
넵 알겠습니다 ㅎㅎ 리뷰 감사합니다!!